R/estimate_infections.R

```diff
@@ -698,15 +698,13 @@ format_fit <- function(posterior_samples, horizon, shift, burn_in, start_date,
   # nolint start
   format_out$samples <- format_out$samples[
     ,
-    type := data.table::fifelse(
-      date > (max(date, na.rm = TRUE) - horizon),
-      "forecast",
-      data.table::fifelse(
-        date > (max(date, na.rm = TRUE) - horizon - shift),
-        "estimate based on partial data",
-        "estimate"
+    type := data.table::fcase(
+      date > (max(date, na.rm = TRUE) - horizon), "forecast",
+      date > (max(date, na.rm = TRUE) - horizon - shift),
+      "estimate based on partial data",
+      is.na(date), NA_character_,
```

Comment:
Is NA_character the right choice for a date (I don't know). Can Na handling not be set directly using `fcase`?

Reply:
It's a string (vs. "forecast", "estimate", etc.). I don't think `fcase` has any `NA` handling sadly (unlike `fifelse`). Perhaps to avoid confunsion perhaps the explicit handling isn't so bad though.